src/main/java/at/ac/tuwien/student/sese2017/xp/hotelmanagement/BuildSearchIndex.java

```diff
@@ -0,0 +1,34 @@
+package at.ac.tuwien.student.sese2017.xp.hotelmanagement;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.hibernate.search.jpa.FullTextEntityManager;
+import org.hibernate.search.jpa.Search;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BuildSearchIndex implements ApplicationListener<ApplicationReadyEvent> {
+
+  @PersistenceContext
+  private EntityManager entityManager;
+
+  /**
+   * Create an initial Lucene index for the data already present in the
+   * database.
+   * This method is called when Spring's startup.
+   */
+  @Override
+  public void onApplicationEvent(final ApplicationReadyEvent event) {
+    try {
+      FullTextEntityManager fullTextEntityManager =
```

Comment:
Explain code in comments. Should be human readable without prior knowledge of the code

Reply:
Done